sentinel.md

```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
 The process of metering is [explained in its own chapter](./metering.md).
 
 The Sentinel is a system contract which means its code is fixed and is part of the genesis or hardfork ewasm is enabled on.
-As a result, changing or updating it requires a hard fork.
+As a result, changing or updating it requires a hard fork. In order to facilitate a more dynamic ecosystem a proposal follows
+here to introduce upgradability for the Sentinel contract.
+
+## Governance (upgradability)
+
+The Sentinel contract is split into two parts:
+- metering injection ("Sentinel")
+- gas cost dataset ("CostTable")
+
+The `CostTable` is queried by the Sentinel during injection. This makes it possible to update either of them with a lower risk.
+
+As a first step, we only propose to have the on-chain ability to update the `CostTable`.
+
+Both of the contracts should conform to the Contract ABI. Their interfaces are given as Solidity `interface`s (which have a
+strict translation to the ABI):
+
+```
+interface CostTable {
+    enum CostKind {
+        BaseInstantiation,
+        MemoryPage,
+        I32Add,
+        I32Sub,
+        I32Mul,
+        I32Div,
+        I32Mod,
+        // TBA
+    }
+
+    // Returns the cost for the given kind.
+    function getCost(CostKind kind) external view returns (uint64);
+
+    // Returns all the costs as an array, in order the enum is defined above.
+    function getAllCosts() etxernal view returns (uint64[]);
+}
+
+interface Sentinel {
+    // Takes a contract as an input, validates and injects metering statements and returns the upadted version.
+    //
+    // Note, the Sentinel contract should also accept an incoming call with the `0x0061736D` selector and process
```

Comment:
I don't understand this comment. Where does the `0x0061736D` selector come from?

Reply:
Wasm header?

Reply:
Aha :) Thanks

Reply:
It is just a trick to avoid changing Hera at this moment 😉